### PR TITLE
docs: add removal of `ClientState` interface functions to migration docs

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 1]
```diff
@@ -55,6 +55,26 @@ Please use the new functions `path.Setup`, `path.SetupClients`, `path.SetupConne
...
...o been removed from the `ClientState` interface: `Initialize`, `Status`, `GetLatestHeight`, `GetTimestampAtHeight`, `VerifyClientMessage`, `VerifyMembership`, `VerifyNonMembership`, `CheckForMisbehaviour`, `UpdateState`, `UpdateStateOnMisbehaviour`,  `CheckSubstituteAndUpdateState` and `VerifyUpgradeAndUpdateState`. ibc-go v9 decouples routing at the `02-client` layer from the light clients' encoding structure (i.e. every light client implementation of the `ClientState` interface is not used anymore to route the requests to the right light client at the `02-client` layer, instead a _light client module_ is registered for every light client type and `02-client` routes the requests to the right light client module based on the client ID). Light client developers must implement the newly introduced `LightClientModule` interface and are encouraged to move the logic implemented in the functions of their light client's implementation of the `ClientState` interface to the equivalent function in the `LightClientModule` interface. The table below shows the equivalence between the `ClientState` interface functions that have been removed and the functions in the `LightClientModule` interface:
```
Comment (colin-axner) [inline docs/docs/05-migrations/13-v8-to-v9.md:58]:
```suggestion
The following functions have also been removed from the `ClientState` interface: `Initialize`, `Status`, `GetLatestHeight`, `GetTimestampAtHeight`, `VerifyClientMessage`, `VerifyMembership`, `VerifyNonMembership`, `CheckForMisbehaviour`, `UpdateState`, `UpdateStateOnMisbehaviour`,  `CheckSubstituteAndUpdateState` and `VerifyUpgradeAndUpdateState`. ibc-go v9 decouples routing at the `02-client` layer from the light clients' encoding structure (i.e. every light client implementation of the `ClientState` interface is not used anymore to route the requests to the right light client at the `02-client` layer, instead a _light client module_ is registered for every light client type and `02-client` routes the requests to the right light client module based on the client ID). Light client developers must implement the newly introduced `LightClientModule` interface and are encouraged to move the logic implemented in the functions of their light client's implementation of the `ClientState` interface to the equivalent function in the `LightClientModule` interface. The table below shows the equivalence between the `ClientState` interface functions that have been removed and the functions in the `LightClientModule` interface:
```

needle in a haystack diffs, "The following functions has also" -> "The following functions have also"

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 2]
```diff
@@ -55,6 +55,26 @@ Please use the new functions `path.Setup`, `path.SetupClients`, `path.SetupConne
...
+|`VerifyMembership`             |`VerifyMembership`           |
+|`VerifyNonMembership`          |`VerifyNonMembership`        |
+|`CheckForMisbehaviour`         |`CheckForMisbehaviour`       |
+|`UpdateState`                  |`UpdateState`                |
+|`UpdateStateOnMisbehaviour`    |`UpdateStateOnMisbehaviour`  |
+|`CheckSubstituteAndUpdateState`|`RecoverClient`              |
+|`VerifyUpgradeAndUpdateState`  |`VerifyUpgradeAndUpdateState`|
+|`ExportMetadata`               |                             |
+
+Please check also the [light client developer guide](../03-light-clients/01-developer-guide/01-overview.md) for more information. The light client module implementation for `07-tendermint` may also be useful as reference.
```
Comment (colin-axner) [inline docs/docs/05-migrations/13-v8-to-v9.md:76]:
```suggestion
|`ExportMetadata`               |                             |
|`ZeroCustomFields`           |                             |

Please check also the [light client developer guide](../03-light-clients/01-developer-guide/01-overview.md) for more information. The light client module implementation for `07-tendermint` may also be useful as reference.
```

**Review by colin-axner — APPROVED** [review]
Thank you :heart: this is excellent

**Review by chatton — APPROVED** [review]
LGTM!

**Pull request merged**